File: backend/src/main/java/com/airline/backend/entity/Booking.java
package com.airline.backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "bookings")
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long bookingId;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "flight_id", nullable = false)
    private Flight flight;

    private int fromSeatNo;
    private int toSeatNo;
    private int noOfSeat;

    private String passenger2;
    private String passenger3;
    private String passenger4;

    @Column(unique = true)
    private String pnr;

    @Enumerated(EnumType.STRING)
    private BookingStatus bookingStatus = BookingStatus.CONFIRMED;
}

File: frontend/src/app/models/booking.ts
export interface Booking {
  bookingId?: number;
  user?: any;
  flight?: any;
  fromSeatNo?: number;
  toSeatNo?: number;
  noOfSeat: number;
  passenger2?: string;
  passenger3?: string;
  passenger4?: string;
  pnr?: string;
  bookingStatus?: 'CONFIRMED' | 'CANCELLED' | 'FLIGHTCANCELLED';
}